packages/lib/src/vis/heatmap/utils.ts

```diff
@@ -138,22 +152,27 @@ export function scaleDomain(
   return [scaleFn(domain[0]), scaleFn(domain[1])];
 }
 
+export function toTextureSafeNdArray(
+  arr: NdArray<NumArray>
+): NdArray<TextureSafeTypedArray> {
+  if (arr.dtype === 'float32' || arr.dtype.startsWith('uint8')) {
+    return arr as NdArray<TextureSafeTypedArray>;
```

Comment:
Seemed over the top to add a type guard to check specifically for these dtypes like I had done before for `float32`...